net: l2: openthread: join thread mcast addrs added to zephyr

frazieje opened the pull request:
Modifies openthread shim layer to automatically join multicast
addresses as they are added to zephyr from openthread, unless the
address is interface-local or link-local. This allows incoming
openthread multicast group messages to avoid being filtered by
zephyr ipv6 recv.

Fixes #31085

Signed-off-by: Joel Frazier <frazieje@gmail.com>

rlubos commented on subsys/net/l2/openthread/openthread_utils.c:
```diff
@@ -269,11 +270,22 @@ void add_ipv6_maddr_to_zephyr(struct openthread_context *context)
+		    !(net_if_ipv6_maddr_is_joined(zmaddr) ||
+		      net_ipv6_is_addr_mcast_iface((struct in6_addr *)
+						   (&maddress->mAddress)) ||
+		      net_ipv6_is_addr_mcast_link_all_nodes((struct in6_addr *)
+							    (&maddress->mAddress)))) {
```
nit: This line exceeds the char limit, I'd suggest to move the cast to the line below like this:
```suggestion
		      net_ipv6_is_addr_mcast_link_all_nodes(
		      		(struct in6_addr *)(&maddress->mAddress)))) {
```
This could apply for `net_ipv6_is_addr_mcast_iface` as well for consistency.

rlubos commented on subsys/net/l2/openthread/openthread_utils.c:
```diff
@@ -269,11 +270,22 @@ void add_ipv6_maddr_to_zephyr(struct openthread_context *context)
 	}
 }
 
+
```
I don't think this empty line is necessary.

rlubos reviewed (commented):
Nice catch, thanks for submitting the PR. I think it's a fine solution for the issue reported, just commented on some nits.

frazieje committed:
net: l2: openthread: join thread mcast addrs added to zephyr

Modifies openthread shim layer to automatically join multicast
addresses as they are added to zephyr from openthread, unless the
address is interface-local or link-local. This allows incoming
openthread multicast group messages to avoid being filtered by
zephyr ipv6 recv.

Fixes #31085

Signed-off-by: Joel Frazier <frazieje@gmail.com>

frazieje commented:
> Nice catch, thanks for submitting the PR. I think it's a fine solution for the issue reported, just commented on some nits.

No problem. I updated the branch to reflect your comments. Thanks for reviewing.
